CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,142 @@
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+We as members, contributors, and leaders pledge to make participation in our
+community a harassment-free experience for everyone, regardless of age, body
+size, visible or invisible disability, ethnicity, sex characteristics, gender
+identity and expression, level of experience, education, socio-economic status,
+nationality, personal appearance, race, caste, color, religion, or sexual
+identity and orientation.
+
+We pledge to act and interact in ways that contribute to an open, welcoming,
+diverse, inclusive, and healthy community.
+
+## Our Standards
+
+Examples of behavior that contributes to a positive environment for our
+community include:
+
+* Demonstrating empathy and kindness toward other people
+* Being respectful of differing opinions, viewpoints, and experiences
+* Giving and gracefully accepting constructive feedback
+* Accepting responsibility and apologizing to those affected by our mistakes,
+  and learning from the experience
+* Focusing on what is best not just for us as individuals, but for the overall
+  community
+
+Examples of unacceptable behavior include:
+
+* The use of sexualized language or imagery, and sexual attention or advances of
+  any kind
+* Trolling, insulting or derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or email address,
+  without their explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Enforcement Responsibilities
+
+Community leaders are responsible for clarifying and enforcing our standards of
+acceptable behavior and will take appropriate and fair corrective action in
+response to any behavior that they deem inappropriate, threatening, offensive,
+or harmful.
+
+Community leaders have the right and responsibility to remove, edit, or reject
+comments, commits, code, wiki edits, issues, and other contributions that are
+not aligned to this Code of Conduct, and will communicate reasons for moderation
+decisions when appropriate.
+
+## Scope
+
+This Code of Conduct applies within all community spaces, and also applies when
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official email address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+This is the default Code of Conduct for all repositories within the `siemens`
```

Comment:
I added this with b5a537da08e38d00eba16f166ad2faae9926dfc7 based on the same text in our security policy. Thought it'd fit best in the existing Scope section but not sure if it's buried too deep in here, WDYT @fgreinacher?

Reply:
I think it's perfect here.

Just wondering whether we could get rid of the chapter above "This Code of Conduct applies within all community spaces, [...], or acting as an appointed
representative at an online or offline event.". It feels a bit too vague for me.

Reply:
Good point, removed that part. It's also less duplicated like this and we already have `If you are unsure whether the Code of Conduct applies feel free to reach out..` below so that should keep things clear.

Reply:
> Community leaders have the right and responsibility to remove, edit

Having the right to edit comments is quite problematic, as the edit is not easily recognizable. IMHO. I vote for removing "edit".

Reply:
@fmoessbauer edits are actually much more visible and auditable. See this thread:

![image](https://github.com/user-attachments/assets/adc8b2a3-1b5a-499b-8022-27d5620bcae3)

I'd keep it as that's the default from the CC CoC and it's also a bit nicer to edit comments than just flat out remove them if they're not completely out of line.

Reply:
Ok, but can you also see the old content? If not, I’m still against allowing edits.

Reply:
Yes, each version is preserved with a little diff. But I'm ok either way, let's see what others say :)

![image](https://github.com/user-attachments/assets/bc3e4669-e4a8-4cc6-8585-789b8ee430a1)

Reply:
I agree that editing is preferable, unless we really want to get fully rid of something (e.g. a leaked secret or a totally misplaced comment).

Reply:
Also note that this also applies to commits which you often inevitably edit when resolving conflicts, etc. – something which is also often hard or impossible to track afterwards (when branches are deleted etc.).  So I think this point needs to stay.